pkg/ccl/sqlproxyccl/conn_migration.go

```diff
@@ -151,9 +164,12 @@ func (f *forwarder) runTransfer() (retErr error) {
 		latencyDur := timeutil.Since(tBegin)
 		f.metrics.ConnMigrationAttemptedLatency.RecordValue(latencyDur.Nanoseconds())
 
+		// When runTransfer returns, it's either the forwarder has been closed,
+		// or the procesors have been resumed.
 		if !ctx.isRecoverable() {
 			log.Infof(logCtx, "transfer failed: connection closed, latency=%v, err=%v", latencyDur, retErr)
 			f.metrics.ConnMigrationErrorFatalCount.Inc(1)
+			f.Close()
```

Comment:
Can we remove the f.Close() calls in this defer? If they are necessary, that is an indication we are not reporting an error to the caller.